clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1179,7 +1196,11 @@ drmemtrace_get_timestamp_from_offline_trace(const void *trace, size_t trace_size
         if (size < 4)
             return DRMEMTRACE_ERROR_INVALID_PARAMETER;
         if (offline_entries[++timestamp_pos].tid.type != OFFLINE_TYPE_THREAD ||
-            offline_entries[++timestamp_pos].pid.type != OFFLINE_TYPE_PID)
+            offline_entries[++timestamp_pos].pid.type != OFFLINE_TYPE_PID ||
+            (offline_entries[++timestamp_pos].extended.type != OFFLINE_TYPE_EXTENDED ||
```

Comment:
Let's make it easy to add more markers in the future with a loop over any markers that happen to be present.

Reply:
It may be error prone to use a loop to iterate until we find the timestamp marker (when do we stop? we might go out of bounds if the timestamp marker isn't there for some reason). 

Also, in the current implementation, the order of these markers is essential to raw2trace. It might be better to retain the extra check.

Reply:
Stop at a non-meta entry (i.e., instruction or load/store).

I'm afraid the next piece of metadata we add will then have to go change this hardcoded check, and the one after that...seems like getting the timestamp doesn't care what other metadata is there.

Reply:
> Wow, looks like a bug: yet another thing that was not finished in the port :(. Please file an issue.

Filed #4446

Reply:
I've added an XXX comment for now. I can send a follow-up PR for this cleanup, instead of packing too many changes in this one.